samples/snippets/csharp/language-reference/keywords/switch/when-clause.cs

```diff
@@ -0,0 +1,91 @@
+// <Snippet1>
+using System;
+
+public abstract class Shape
+{
+   public abstract double Area { get; }
+   public abstract double Circumference { get; } 
+}
+
+public class Rectangle : Shape
+{
+   public Rectangle(double length, double width) 
+   {
+      Length = length;
+      Width = width; 
+   }
+
+   public double Length { get; set; }
+   public double Width { get; set; }
+   
+   public override double Area
+   { 
+      get { return Math.Round(Length * Width,2); } 
+   } 
+   
+   public override double Circumference 
+   {
+      get { return (Length + Width) * 2; }
+   }
+}
+
+public class Square : Rectangle
+{
+   public Square(double side) : base(side, side) 
+   {
+      Side = side; 
+   }
+
+   public double Side { get; set; }
+}
+
+public class Example
+{
+   public static void Main()
+   {
+      Shape sh = null;
+      Shape[] shapes = { new Square(10), new Rectangle(5, 7),
+                         sh, new Square(0) };
+      foreach (var shape in shapes)
+         ShowShapeInfo(shape);
+   }
+
+   private static void ShowShapeInfo(Shape sh)
+   {
+      switch (sh)
+      {
+         case Shape shape when shape == null:
```

Comment:
this won't match. That's because the type pattern does not match null.

I would leave this in the example, but point out that it doesn't work.

To find a null, you use the constant null pattern:

`case null:`

I think that if you had both case expressions, and explained which worked and why it would really illuminate the concept for readers.